docs/PFC-test-plan.md

```diff
@@ -0,0 +1,200 @@
+mechanism for multiple CoS, thus enabling coexistence of flows which
+needs flow control with other flows which performs better without it.
+Upto 8 CoSs can be used. PFC uses the same 64Byte MAC Control frame
+format that PAUSE frames do. It has a two byte Class-Enable vector to
+mention the CoS value for which the PAUSE should apply. As PFC acts
+independently on eight different CoSs, the frame describes the PAUSE
+duration for each CoS. The pause duration for each priority is a 2-byte
+value that expresses time as a number of quanta, where each quanta
+represents the time needed to transmit 512 bits at the current network
+speed. For user data traffic, the CoS maps to either CoS values defined
+in 802.1Q VLAN tag or the IP DSCP values.
+
+SONiC has two lossless priorities: 3 and 4, by default. It is to be
+noted that only the lossless priorities can react to or generate PFC
+frames. In other words, PFC frames should not have any impact on traffic
+on lossy priorities. Packets with Differentiated Services Code Point
+(DSCP) 3 and 4 are mapped to priority 3 and 4, respectively.
+
+### Scope
+
+The purpose of this test plan is to test the PFC PAUSE processing behavior of a SONiC DUT and its capability to pause or un-pause traffic with right priorities. The test assumes all necessary configuration is already pre-configured on the SONIC DUT before test runs.
+
+### Testbed
+
+```
++-------------+      +--------------+      +-------------+
+| Keysight TX |------|   SONiC DUT  |------| Keysight RX |
++-------------+      +--------------+      +-------------+
+
+Keysight ports are connected with SONiC switch as shown in the illustration above.
+```
+## Setup configuration
+
+### DUT configuration
+
+- PFC watch dog is disabled
+
+### Keysight configuration
+
+- All Keysight ports should have the same bandwidth capacity.
+
+## Test cases
+
+### Test Case #1 - PFC PAUSE with single lossless priority
+
+#### Test Objective
+
+Verify DUT processes the PFC PAUSE frame with single lossless CoS
+properly.
+
+#### Test Configuration
+
+- On SONiC DUT configure a single lossless CoS value Pi. (0 \<= i \<= 7).
+- Configure following traffic items on the Keysight device:
+  1. Test data traffic: A traffic item from the Keysight Tx port to
+        the Keysight Rx port with lossless priority (DSCP value == Pi).
+        Traffic should be configured with 50% of line rate.
+  2. Background data traffic: A traffic item from the Keysight Tx
+        port to the Keysight Rx port with lossy priorities (DSCP value
+        != Pi). Traffic should be configured with 50% of line rate.
+  3. PFC PAUSE storm: Persistent PFC pause frames from the Keysight
+        Rx port to the Keysight Tx port. The priorities of PFC pause
+        frames should be same as that of 'Test data traffic'. And the
+        inter-frame transmission interval should be smaller than
+        per-frame pause duration.
+
+#### Test Steps
+
```

Comment:
The description of test steps for all the test cases should be consistent with comments in https://github.com/Azure/sonic-mgmt/pull/1979/files. Many 'CoS' should be modified to 'priority'.

Reply:
There is not many changes on test scripts vs test scripts . However this is an ongoing process and we need to allign test script vs test plan doc. In order to track that we have added reversion as once we complete scripting of the test cases , we will modify the test cases on this planned in subsequent version.